e2e/awslogs_test.go

```diff
@@ -47,12 +59,193 @@ var testAwslogs = func() {
+				LogStreamName: aws.String(testAwslogsStream),
+			})
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
 		})
 		ginkgo.AfterEach(func() {
-			cleanupAwslogs(cwClient, testAwslogsGroup, testAwslogsStream)
+			deleteLogGroup(cwClient, testAwslogsGroup)
+			deleteLogGroup(cwClient, nonExistentAwslogsGroup)
 		})
-		ginkgo.It("should send logs to awslogs log driver", func() {
+		ginkgo.It("should send logs to awslogs log driver with existing log group and non-existent log stream "+
+			"when the configs are default", func() {
+			testLog := testLogPrefix + uuid.New().String()
+			args := map[string]string{
+				logDriverTypeKey:              awslogsDriverName,
+				containerIDKey:                testContainerID,
+				containerNameKey:              testContainerName,
+				awslogsCredentialsEndpointKey: testAwslogsCredentialEndpoint,
+				awslogsRegionKey:              testAwslogsRegion,
+				awslogsGroupKey:               testAwslogsGroup,
+				awslogsStreamKey:              nonExistentAwslogsStream,
+				awslogsEndpointKey:            testAwslogsEndpoint,
+			}
+			creator := cio.BinaryIO(*Binary, args)
+			err := sendTestLogByContainerd(creator, testLog)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			err = validateTestLogsInAwslogs(cwClient, testAwslogsGroup, nonExistentAwslogsStream, testLog)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		})
+		ginkgo.It("should send logs to awslogs log driver with respecting datatime format and ignoring multiline pattern "+
+			"when both datetime format and multiline pattern are set", func() {
+			firstLineLog := "[May 01, 2017 19:00:01] " + testLogPrefix + uuid.New().String()
+			secondLineLog := "[May 01, 2017 19:00:04] " + testLogPrefix + uuid.New().String()
+			thirdLineLog := fmt.Sprintf("%s %s", testAwslogsMultilinePattern, testLogPrefix+uuid.New().String())
+			forthLineLog := fmt.Sprintf("%s %s", testAwslogsMultilinePattern, testLogPrefix+uuid.New().String())
+			args := map[string]string{
+				logDriverTypeKey:              awslogsDriverName,
+				containerIDKey:                testContainerID,
+				containerNameKey:              testContainerName,
+				awslogsCredentialsEndpointKey: testAwslogsCredentialEndpoint,
+				awslogsRegionKey:              testAwslogsRegion,
+				awslogsGroupKey:               testAwslogsGroup,
+				awslogsStreamKey:              nonExistentAwslogsStream,
+				awslogsEndpointKey:            testAwslogsEndpoint,
+				awslogsMultilinePatternKey:    "^" + testAwslogsMultilinePattern,
+				awslogsDatetimeFormatKey:      testAwslogsDatetimeFormat,
+			}
+			creator := cio.BinaryIO(*Binary, args)
+			err := sendTestLogByContainerd(creator, fmt.Sprintf("%s\n%s\n%s\n%s", firstLineLog, secondLineLog, thirdLineLog, forthLineLog))
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			err = validateTestLogsInAwslogs(cwClient, testAwslogsGroup, nonExistentAwslogsStream, fmt.Sprintf("%s\n", firstLineLog))
```

Comment:
should we validate all lines here instead of just the first?

Reply:
Good call. Unfortunately I identified a weird issue during this check. I opened an issue and added TODO to track it. https://github.com/aws/shim-loggers-for-containerd/issues/78